uber/decorators.py

```diff
@@ -110,6 +110,42 @@ def charge(self, session, payment_id, stripeToken, stripeEmail='ignored', **igno
     return charge
 
 
+def cached(func):
+    func.cached = True
+    return func
+
+
+def cached_page(func):
+    from sideboard.lib import config as sideboard_config
+    innermost = get_innermost(func)
+    func.lock = RLock()
+    @wraps(func)
+    def with_caching(*args, **kwargs):
+        if hasattr(innermost, 'cached'):
+            fpath = os.path.join(sideboard_config['root'], 'data', func.__module__ + '.' + func.__name__)
+            with func.lock:
+                if not os.path.exists(fpath) or datetime.now().timestamp() - os.stat(fpath).st_mtime > 60 * 15:
```

Comment:
If we ended up doing this more long-term, we could make this configurable, either in our file config or as a parameter to the ``@cached`` decorator.  For now, 15 minutes seems fine.